amazon-kinesis-client/src/test/java/software/amazon/kinesis/coordinator/DiagnosticEventsTest.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package software.amazon.kinesis.coordinator;
+
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import software.amazon.kinesis.leases.Lease;
+import software.amazon.kinesis.leases.LeaseBuilder;
+import software.amazon.kinesis.leases.LeaseCoordinator;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@Slf4j
+@RunWith(MockitoJUnitRunner.class)
+public class DiagnosticEventsTest {
+    @Mock
+    private ThreadPoolExecutor executor;
+    @Mock
+    private LeaseCoordinator leaseCoordinator;
+    @Mock
+    private DiagnosticEventHandler defaultHandler;
+
+    private DiagnosticEventHandler customHandler;
+    private boolean wasCustomHandlerInvoked;
+
+    private Throwable throwable;
+
+    private int activeThreadCount;
+    private int corePoolSize;
+    private int largestPoolSize;
+    private int maximumPoolSize;
+
+    private SynchronousQueue<Runnable> executorQueue;
+    private Collection<Lease> leaseAssignments;
+
+    @Before
+    public final void setup() {
+        customHandler = new CustomHandler();
+        wasCustomHandlerInvoked = false;
+
+        throwable = new TestRejectedTaskException();
+
+        activeThreadCount = 2;
```

Comment:
These values don't seem to change, they can be made finals

Reply:
Made these values final